### PR TITLE
Add ability to upload profile photo with s3

### PR DESCRIPTION
**Pull request by NiallJoeMaher**
# ✨ Codu Pull Request 💻

## Pull Request details:
- Adds Dockerfile to test build on AWS
- Removes use of  Nextjs Image when using bucket for profile photo until we have a domain attached 

## Any Breaking changes:
To upload photos locally you need to setup an s3 bucket

Screenshot 2022-12-31 at 16 17 21

### CODE REVIEW

File: server/trpc/router/profile.ts [hunk 1]
```diff
@@ -16,6 +24,43 @@ export const profileRouter = router({
...
+      const extension = type.split("/")[1];
+
+      if (!["jpg", "jpeg", "gif", "png", "webp"].includes(extension)) {
+        throw new TRPCError({
+          code: "BAD_REQUEST",
+          message: "Invalid file format.",
+        });
+      }
+
+      if (size > 8388608) {
```
Comment (pkspyder007) [inline server/trpc/router/profile.ts:53]: can we keep this somewhere as a constant?

File: server/trpc/router/profile.ts [hunk 2]
```diff
@@ -16,6 +24,43 @@ export const profileRouter = router({
...
+      });
+      return profile;
+    }),
+  getUploadUrl: protectedProcedure
+    .input(uploadPhotoUrlSchema)
+    .mutation(async ({ ctx, input }) => {
+      const { size, type } = input;
+      const extension = type.split("/")[1];
+
+      if (!["jpg", "jpeg", "gif", "png", "webp"].includes(extension)) {
```
Comment (pkspyder007) [inline server/trpc/router/profile.ts:46]: making out a helper utility make sense for re-using such logic

File: server/trpc/router/profile.ts (hunk 1 above)
Comment (NiallJoeMaher) [reply server/trpc/router/profile.ts:53]: I'll refactor this later, I think there is a more elegant way to do this with the s3 client I'm just still learning it.

File: server/trpc/router/profile.ts (hunk 2 above)
Comment (NiallJoeMaher) [reply server/trpc/router/profile.ts:46]: I'll pull it out if it becomes helpful in another part. I don't see it being used anywhere else for the moment 😁

**Review by pkspyder007 — APPROVED** [review]

**Pull request merged**